app/controllers/user_applications_controller.rb

```diff
@@ -34,7 +34,6 @@ def new
     if @user_information.nil?
       # Verify that the user has a user information
       session[:return_to] = new_user_application_url
-      session[:origin] = URI(request.referer).path unless request.referer.nil?
       redirect_to new_user_information_url and return
```

Comment:
might not need the "return" here as redirect_to should halt execution. Might be wrong.